src/main/java/com/fasterxml/jackson/databind/introspect/POJOPropertiesCollector.java

```diff
@@ -653,7 +690,8 @@ protected void _addCreatorParam(Map<String, POJOPropertyBuilder> props,
             // Also: if this occurs, there MUST be explicit annotation on creator itself
             JsonCreator.Mode creatorMode = _annotationIntrospector.findCreatorAnnotation(_config,
                     param.getOwner());
-            if ((creatorMode == null) || (creatorMode == JsonCreator.Mode.DISABLED)) {
+            if ((creatorMode == null || creatorMode == JsonCreator.Mode.DISABLED)
+                    && defaultParameterName.isEmpty()) {
```

Comment:
Should add a comment explaining logic for "defaultParameterName" somewhere (what/why?)